cli-output/src/cli_output.rs

```diff
@@ -66,28 +65,90 @@ pub enum OutputFormat {
     DisplayVerbose,
 }
 
-impl OutputFormat {
-    pub fn formatted_string<T>(&self, item: &T) -> String
-    where
-        T: Serialize + fmt::Display + QuietDisplay + VerboseDisplay,
-    {
-        match self {
-            OutputFormat::Display => format!("{item}"),
+/// CLI supports outputting processing results in the JSON format.  Objects that can be output in
```

Comment:
I've modified this comment, but much less than you are suggesting.
Below is the detail explanation of why.

---

> To me it would be easier to comprehend this text if it were organized in a different way: each paragraph starting from the main idea and later details. So I don't need to parse the whole comment if I don't really need this particular information. Like:
 
```rust
/// This trait enables JSON output for CLI, intended for objects supporting JSON mode.
```

I've removed the first sentence of the first paragraph, and this seems to be almost identical to the second sentence I wrote:
```rust
/// Objects that can be output in the JSON mode should implement this trait.
```
I'll keep it, if that is OK.

---

```rust
/// Alongside [`FormatForMode`], this trait is used to ease implementation... In contrast
/// [`FormatForMode`] is for human-readable output and, hence, requires a more nuanced
/// implementation.
```

I must admit that I do not understand how is this more clear.
The first sentence that I have says that `FormatForJson` is used by `FormatForMode`.
Which, I think, is an important detail.  But you suggest I remove it?
And the last sentence you are suggesting, to me, is just a combination of two sentences I wrote:

```rust
/// On the other hand [`FormatForMode`] covers output for a human consumption.  And the
/// later is often overwritten in a more nuanced way, so it can not be generated automatically.
```

---

```rust 
/// Unlike [`serde::Serialize`], this trait is object-safe, making it more ergonomic. Since it
/// produces complete `String`s, the potential performance gains from inlining `format_for_json` are
/// likely negligible.
```

This seems to be a rephrasing of what I wrote, but some details are missing:

```rust
/// This trait is also very similar to [`serde::Serialize`], but it is object safe, making it a bit
/// more ergonomic in certain cases.  As we are producing whole `String`s, inlining of
/// `format_for_json`, that monomorphization can provide, most likely, have no real impact on the code
/// performance.
```

I think that the fact that it is similar to `Serialize` is quite important, and this is why I put it first.
But you remove it and put the difference with `Serialize` first.
If I follow your suggestion of putting the important idea first, I think my version is closer to what I am trying to say.

By removing the "in certain cases" part, from the "making it more ergonomic in certain cases", you change the meaning.
`Serialize` can be more ergonomic if you do not care for object safety.
And wise versa if you do.  So it is not a clear advantage.
And this is why I am talking about object safety first.

Monomorphization is the reason that `Serialize` can be faster, as the compiler will write a different version of the function for each type, which in turn enables better inlining.

I think if I remove the mention of "monomorphization" it might not be as easy to understand why am I talking about inlining in the first place.
As, it is still possible to inline `format_for_json` even as defined by this trait.

Reply:
I don't insist on the text I've provided, it was for illustration purposes, with `...` for the additional details.
My general comment was that it would be nice if I read the first sentence of the comment and understand immediately if I need to read further or it is irrelevant for whatever I'm looking for. The updated first sentence looks good in this respect.